cards-template1b.js

```diff
@@ -1,26 +1,75 @@
 // Error-detecting version
```

Comment:
You've got a fair amount of this working well, but after running some of the later (Number.isNan) assertions, we run into some problems. When checking for validity of your arguments in these functions, remember that we are not only checking to see if the number is between 0-51, but also if the argument given is a number at all. 

While thinking about that, consider also whether you could make a helper function that checks the validity of the argument given to your functions, that replaces copying/pasting the same line(s) of code into each function.

You're on the right path here, keep pressing forward!